packages/datadog-plugin-kafkajs/src/index.js

```diff
@@ -0,0 +1,76 @@
+'use strict'
+
+function createWrapProducer (tracer, config) {
+  return function wrapProducer (createProducer) {
+    return function producerWithTrace () {
+      const serviceName = config.service || `${tracer._service}-kafka`
+      const createdProducer = createProducer.apply(this, arguments)
+
+      const send = createdProducer.send
+
+      const tags = {
+        'service.name': serviceName,
+        'span.kind': 'producer',
+        'span.type': 'queue',
+        component: 'kafka'
+      }
+
+      createdProducer.send = tracer.wrap('kafka.producer.send', { tags }, function (...args) {
+        const { topic, messages } = args[0]
+        const currentSpan = tracer.scope().active()
+
+        currentSpan.addTags({
+          'resource.name': `produce to ${topic}`,
+          'kafka.topic': topic,
+          'kafka.batch.size': messages.length
+        }
+        )
+        return send.apply(this, args)
+      })
+
+      return createdProducer
+    }
+  }
+}
+
+function createWrapConsumer (tracer, config) {
+  return function wrapProcessEachMessage (Consumer) {
+    return function processEachMessageWithTrace () {
+      const consumer = Consumer.apply(this, arguments)
+      const run = consumer.run
+
+      consumer.run = async function ({ eachMessage, ...args }) {
+        // return the promise
+        return run({
+          eachMessage: tracer.wrap('kafkajs.consumer', {}, eachMessage),
```

Comment:
Similar to the producer, the span name should be something like `kafka.consume` or `kafka.receive` depending on Kafka terminology.

Reply:
There should also be the additional tags similar to the producer, for example `span.kind`, `component`, etc.